datalad/support/annexrepo.py

```diff
@@ -3408,10 +3408,10 @@ class AnnexJsonProtocol(WitlessProtocol):
     proc_out = True
     proc_err = True
 
-    def __init__(self, done_future, total_nbytes=None):
+    def __init__(self, total_nbytes=None):
```

Comment:
I wonder if we should avoid breaking API for a cycle here and still allow for `done_future=None` and issue DeprecationWarning if specified?

Reply:
Good point. I am not fully aware, where it is used and assumed we use it only internally. Do we expose it externally, i.e. do we document it and encourage its use? To me either way is fine.

Reply:
well -- wonders of Python that as long as it is not `_`'ed somewhere in the namespace path to it -- it is "public".  But it is so specific and I do not see hits for https://github.com/search?q=AnnexJsonProtocol&type=code at least outside of datalad, and tests for extensions are green -- I guess we could just jump and not care about deprecation cycle.

Reply:
Deprecation cycle added